javascripts/discourse/initializers/header-search.js

```diff
@@ -13,7 +13,13 @@ export default {
           {{header-topic-info attrs=attrs}}
         {{else}}
           {{#unless this.site.mobileView}}
-            {{floating-search-input attrs=attrs}}
+            {{#if this.site.siteSettings.login_required}}
```

Comment:
I think maybe we can just use the `(and this.site.siteSettings.login_required this.currentUser)` helper for this?